src/nvim/api/private/helpers.c

```diff
@@ -644,10 +644,8 @@ static Object vim_to_object_rec(typval_T *obj, PMap(ptr_t) *lookup)
 
   switch (obj->v_type) {
     case VAR_STRING:
-      if (obj->vval.v_string != NULL) {
-        rv.type = kObjectTypeString;
-        rv.data.string = cstr_to_string((char *) obj->vval.v_string);
-      }
+      rv.type = kObjectTypeString;
+      rv.data.string = cstr_to_string((char *) obj->vval.v_string);
```

Comment:
Nah, I did at one point submit a PR to prevent `jobstart` from segfaulting due to a `NULL` string, but I never tracked down from where it came. Personally, I think it's safer to assume that a string may be `NULL` unless we're 100% sure that the entire codebase has been sanitized.